main/datalog.go

```diff
@@ -216,16 +216,17 @@ func bulkInsert(tbl string, db *sql.DB) (res sql.Result, err error) {
 
 	batchVals := insertBatchIfs[tbl]
 	for len(batchVals) > 0 {
-		i := int(0) // Maximum of 25 rows per INSERT statement.
+		i := int(0)
 		stmt := ""
 		vals := make([]interface{}, 0)
-		querySize := uint64(0) // Size of the query in bytes.
-		for len(batchVals) > 0 && i < 1000 && querySize < 750000 {
+		querySize := uint64(0)                                   // Size of the query in bytes.
+		for len(batchVals) > 0 && i < 30 && querySize < 750000 { // Maximum of 1000? columns for INSERT on default sqlite compile. trafficData is limiting function with 31 parameters.
```

Comment:
I just woke up, so if I'm not making sense...

How would what work?  The B-tree method?  The code snippet you post here?

If you mean the B+Tree, you can use the source and compile it into 
stratux and add a record for each timestamp.  If each timestamp data is 
simply output into a text file, one doesn't need any extra code. It can 
all be handled in post-processing which is how I was planning on doing 
it.  One can, of course, use fixed position of each field (lots of empty 
characters sometimes) or one can use field delimiters (commas, quote 
marks, etc) but that incurs even more overhead in the file if most 
fields are populated every time.  To save on SD card space, each 
timestep record (or batch of records) could be compressed in stratux, 
and decompressed during post-processing.  The content of the data 
records will dictate whether that is efficient or not, but for ASCII, it 
usually saves a great deal of disk/SD card space.  Post-processing would 
require a separate executable outside the stratux code itself but that 
is the most efficient method.

I'm a bit confused here, though.  I am at a loss as to the 1,000+ 
columns (fields?) for  each timestamp.  Are there really that many?

Terms: in a 2D database (aka table aka set), columns aka fields are the 
data items that go across, while records aka entries, aka entities go 
down.  As  in one record per timestamp and each record contains the data 
relevant to that, and only that, timestamp. Example: a database (table) 
of stratux users would look something like this --
columns aka fields defined with all ascii text characters as
LastName (10 chars)  FirstName (10 chars) DateStarted  (8 chars) 
TimeStarted (4 chars HHMM) Platform (3 chars code) and so on, yielding 
one row/record, going across the 2D database..
Then, each person is a separate row/record/entity going down the 2D 
database,
SQL has the tables, subtables, and so on, but for  our purposes in this 
example, the set of all stratux users is the total table aka database as 
there is only one "table" in our desired info.  (This example uses easy 
fixed length records just so we can define out terms.)

Let the stratux code just store the timestamp data into a (flat)file, 
and all the heavy lifting is done in post-processing. That way, each 
person can use the database engine they like and they just import that 
data whenever they choose.  Simple.

Timestamps can be unique per record or if there are multiple entries per 
timestamp, non-unique.  If non-unique, we get into 3D, 4D, and so  on 
database, or varying number of entries of each field per timestamp which 
would be an entirely different can of worms.  (I *like* things smple.)  
:)  But then, you'll need a hierarchical database engine instead of a 
relational one.  But that's all the job of the post-processor.

Am I rambling yet?

Did I answer the question?

Skypuppy



On 04/03/2016 01:17 PM, cyoung wrote:
>
> In main/datalog.go 
> <https://github.com/cyoung/stratux/pull/365#discussion_r58314775>:
>
> >  		stmt := ""
> >  		vals := make([]interface{}, 0)
> > -		querySize := uint64(0) // Size of the query in bytes.
> > -		for len(batchVals) > 0 && i < 1000 && querySize < 750000 {
> > +		querySize := uint64(0)                                   // Size of the query in bytes.
> > +		for len(batchVals) > 0 && i < 30 && querySize < 750000 { // Maximum of 1000? columns for INSERT on default sqlite compile. trafficData is limiting function with 31 parameters.
>
> @skypuppy <https://github.com/skypuppy> - how would this work?
> The idea behind this is to correlate all of the data that we have, let 
> it be properly timestamped, and have it work across the multiple 
> hardware profiles we have now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/cyoung/stratux/pull/365/files/a546f18750e1ebf0f5d5086889ab3513ed0deb3a#r58314775>
>